### ISSUE TITLE
Mapping mouse to analog stick?

### ISSUE THREAD

**Issue by Krusher97** [m64p/mupen64plus-gui#28]
According to this : http://mupen64plus.org/wiki/index.php?title=ControllerSetup#Using_a_Mouse
The Mupen64plus input plugin supports mapping mouse inputs for analog sticks and various other buttons. I couldn't find a way to do this in the GUI.

**Comment by loganmc10** [comment]
> The Mupen64plus input plugin supports mapping mouse inputs for analog sticks and various other buttons. 

It does, but it uses SDL. This GUI uses Qt, and in order for SDL mouse input to work, it needs an SDL window, the same as for keyboard input. The mupen64plus API has an API for injecting keyboard input (M64CMD_SEND_SDL_KEYDOWN/M64CMD_SEND_SDL_KEYUP), so that I can capture the keyboard input using Qt, then convert it to an SDL keycode, and send it to mupen64plus.

There is no such API for mouse input, so there is no way for me currently to capture mouse input, changes would need to be made to the mupen64plus API, and potentially the input plugin (I haven't looked at the code closely there).

**Comment by Krusher97** [comment]
So is there no way for me to just directly edit an ini to get mouse capabilities?

**Comment by loganmc10** [comment]
> So is there no way for me to just directly edit an ini to get mouse capabilities?

No, the GUI doesn't support it. It is not able to capture mouse input and pass it to the emulator.

**Comment by SigmaVirus** [comment]
@Krusher97 
I think you technically can use https://github.com/AntiMicro/antimicro in order to achieve what you want.

I use it for Mednafen and Citra and AntiMicro works fantastic.
